### PR TITLE
Add "include synthetic parameters" to mapping stats & save options

### LINKED ISSUE #204: Mapping stats: ignore synthetic methods in counting
Currently mapping stat counts lambdas as unmapped. It'd be sweet if there is an option to ignore synthetic fields/methods.

### PR DESCRIPTION
**Pull request by YanisBft**
Also updated French translation
Closes #204 (even though the fix itself was pushed a while ago)

### CODE REVIEW

**Comment by Runemoro** [comment]
The bug you linked isn't asking to include synthetics, it's saying synthetics are being incorrectly *included* (that the check isn't working properly)...

Why would anyone want to include synthetics, since those can't be mapped?

**Comment by YanisBft** [comment]
The issue also asked for an option to ignore it or not. And synthetics can be mapped:
image
The synthetic method in red on the screenshot isn't counted in stats, but would be if you select the option added in this PR

**Comment by Runemoro** [comment]
But not the method name, only the parameter names. It should only count parameters to lambdas (but not bridges).

The issue is asking for an option to ignore it, **because the current check isn't working**. Making the current check optional only makes it worse.

**Comment by YanisBft** [comment]
@Runemoro as there was a misunderstanding on my side, reworded to be an option to include synthetic parameters

**Review by modmuss50 — APPROVED** [review]

File: enigma-swing/src/main/java/cuchaz/enigma/gui/config/UiConfig.java [hunk 1]
```diff
@@ -266,6 +266,22 @@ public static void setLastSelectedDir(String directory) {
 		swing.data().section("File Dialog").setString("Selected", directory);
 	}
 
+	public static String getLastTopLevelPackage() {
+		return swing.data().section("Mapping Stats").getString("Top-level package").orElse("");
```
Comment (2xsaiko) [inline enigma-swing/src/main/java/cuchaz/enigma/gui/config/UiConfig.java:270]: Please use title case for these ("Top-level package" -> "Top-Level Package")

File: enigma/src/main/resources/lang/en_us.json [hunk 2]
```diff
@@ -28,6 +28,7 @@
 	"menu.file.stats": "Mapping Stats...",
 	"menu.file.stats.title": "Mapping Stats",
 	"menu.file.stats.top_level_package": "Top-level package:",
+	"menu.file.stats.synthetic_parameters": "Include synthetic parameters",
```
Comment (2xsaiko) [inline enigma/src/main/resources/lang/en_us.json:31]: Same here, use title case (while you're at it, change the one for top-level package too)

**Review by 2xsaiko — CHANGES_REQUESTED** [review]
Code looks good, some minor style things

**Comment by YanisBft** [comment]
Fixed capitalization

**Review by Runemoro — APPROVED** [review]

**Review by 2xsaiko — APPROVED** [review]

**Pull request merged**